File: R/statistics_with_R/03_R_Enviroment/Script_Files/05_reshapingData.R
data = read.delim('/home/atrides/Desktop/Applied-Statistics-with-R-master/statistics_with_R/03_R_Enviroment/Data_Files/Honeymoon Period.dat', header=TRUE)

head(data)
names(data)
# converting wide df to long df
satisfactionStacked<-stack(data,select=c("Satisfaction_Base","Satisfaction_6_Months","Satisfaction_12_Months","Satisfaction_18_Months"))


head(satisfactionStacked)


# converting long to wide
satisfactionUnstacked <- unstack(satisfactionStacked, values~ind)
satisfactionUnstacked


# install.packages("reshape")
library(reshape)

restructuredData<-melt(data, id = c("Person", "Gender"), measured = c("Satisfaction_Base", "Satisfaction_6_Months",
                                                                      "Satisfaction_12_Months", "Satisfaction_18_Months"))

head(restructuredData)


wideData<-cast(restructuredData, Person + Gender ~ variable, value = "value")
wideData

File: Python/statistics_with_Python/03_Python_Enviroment/Script_Files/05_reshapingData.py
import pandas as pd
import numpy as np



data = pd.read_csv('/home/atrides/Desktop/R/statistics_with_Python/03_Python_Enviroment/Data_Files/Honeymoon Period.dat', sep='\s+')


print(data.head())


print(data.columns)


# wide data to long data
melted_df = pd.melt(data,id_vars=['Person','Gender'], value_vars=['Satisfaction_Base','Satisfaction_6_Months','Satisfaction_12_Months','Satisfaction_18_Months'], 
       var_name='satisfaction_after_time')
print(melted_df.head(10))


# long data to wide data
unmelted_df = pd.pivot_table(melted_df, values='value', index=['Person'],

                    columns=['satisfaction_after_time'], dropna=False)
_ = melted_df[melted_df['satisfaction_after_time']=='Satisfaction_Base']['Gender']
unmelted_df.columns = ['Satisfaction_Base','Satisfaction_6_Months', 'Satisfaction_12_Months','Satisfaction_18_Months']
unmelted_df['Person'] = unmelted_df.index
unmelted_df.reset_index(drop=True, inplace=True)
unmelted_df['Gender'] = _


# data converted from long to wide
print(unmelted_df)



